apiserver/facades/agent/resourceshookcontext/register.go

```diff
@@ -25,42 +21,14 @@ func Register(registry facade.FacadeRegistry) {
 }
 
 // newStateFacade provides the signature to register this resource facade
-func newStateFacade(ctx facade.ModelContext) (*UnitFacade, error) {
-	authorizer := ctx.Auth()
-	st := ctx.State()
+func newStateFacade(modelCtx facade.ModelContext) (*UnitFacade, error) {
+	authorizer := modelCtx.Auth()
 
 	if !authorizer.AuthUnitAgent() && !authorizer.AuthApplicationAgent() {
```

Comment:
This is not code you're changing, but it seems weird that we don't care if the unit agent or the application agent is the one authoirzed, we could let an unathorized unit through if its application was authorized (if such a thing can happen)

Reply:
Seems confusing, but I guess we can just leave it as it is.